### ISSUE TITLE
Reduce memory footprint: TinyCore Linux, Alpine Linux, Micropython

### ISSUE THREAD

**Issue by mfp20** [QubesOS/qubes-issues#5696]
**The problem you're addressing (if any)**
QubesOS major disadvantage is resource consumption, and above all RAM utilization. This create problems for most of the potential users out there, especially the ones using laptops; considering a couple of (4GB+) internet browsing machines, those on laptops are often limited to 3-4 app VMs at max, and very little compartmentalization advantages. By using extreme resource optimized linux flavors it is possible to mitigate this problem and unleashing Qubes' full potential. The same applies for qubes specific tools written in python: micropython runs on less than 1MB of RAM.

**Describe the solution you'd like**
**Ideally** moving dom0, system VMs (net, usb), and internet browsing machines, to a musl/uclibc based distro (ex: Alpine Linux) would do the trick, but usually moving out from glibc brings to a subset of available packages, more effort to port any exotic package, and limited support from upstream distribution teams because of a smaller users number for those less popular distributions.
Moreover, Alpine linux can run from a squashfs read-only fs, overlay-ed with a read-write fs (alongside a traditional rw install); it's way more flexible (and safe) than current /rw method.
Another good optimization could be to slowly shift qubes specific tools to Micropython and get rid of libvirt (and its python tools).
But as a start **introducing a TinyCore template would be enough** to start experimenting and give some more juice to low-RAM systems.

**Where is the value to a user, and who might that user be?**
Low-spec systems, laptops, are the ones to benefit most; but I'd benefit too despite my 16 cores and 64GB of RAM, as I'd be able to increase the compartmentalization level multiplying the number of concurrent VMs.

**Describe alternatives you've considered**
Honestly I haven't been searching for alternatives in recent times. In the past there used to be 2-3 tiny Linux distros (ex: something like "Linux On Floppy"). Some of those could be alive.

**Comment by ben-grande** [comment]
I agree with the goal, but this issue is not being currently referenced by the alpine issue:

- #7323

Which was created after this issue. About using TinyCore or Micropython, I think that separate issues should be opened, and in the case of Micropython, needs some study if it works with qubes packages.

Closing as duplicate.
